### PR TITLE
Add request rspec for users and posts controllers

### PR DESCRIPTION
**Pull request by Bantealem**
- [x] In this pull request I have done the following 
- [x]  I have created a Request spec file for all my controllers.
If response status was correct.
If a correct template was rendered.
If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/users_spec.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      expect(response).to render_template(:index)
+    end
+    it 'Should return the correct placeholder' do
+      expect(response.body).to include('Users')
+    end
+  end
+
+  describe 'GET/show' do
+    before :each do
+      get '/users'
```
Comment (omar-muhamad) [inline spec/requests/users_spec.rb]: - [ ] Kindly get the correct route for the show controller which is `/users/:id`.

File: spec/requests/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    end
+    it 'is a success' do
+      expect(response).to have_http_status(:success)
+    end
+    it 'Should return the correct placeholder' do
+      expect(response.body).to include('Users')
+    end
+  end
```
Comment (omar-muhamad) [inline spec/requests/users_spec.rb]: - [ ] Please add a test that ensures the correct view template is rendered.
[Hint]: like that:
```ruby
it 'renders the correct template' do
    expect(response).to render_template(:show)
end
```

File: spec/requests/posts_spec.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      expect(response).to render_template(:index)
+    end
+    it 'Should return the correct placeholder' do
+      expect(response.body).to include('Posts')
+    end
+  end
+
+  describe 'GET/show' do
+    before :each do
+      get '/users/:user_id/posts/'
```
Comment (omar-muhamad) [inline spec/requests/posts_spec.rb]: - [ ] Same here, Kindly get the correct route for the post-show controller which is ` /users/:user_id/posts/:id`.

File: spec/requests/posts_spec.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    end
+    it 'is a success' do
+      expect(response).to have_http_status(:success)
+    end
+    it 'Should return the correct placeholder' do
+      expect(response.body).to include('Posts')
+    end
+  end
```
Comment (omar-muhamad) [inline spec/requests/posts_spec.rb]: - [ ] Same here,  Please add a test that ensures the correct view template is rendered.

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Bantealem,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: spec/requests/posts_spec.rb (hunk 4 above)
Comment (Bantealem) [reply spec/requests/posts_spec.rb]: @Omar-Muhamad 
Thank you for your constructive feedback

**Review by codecaiine — APPROVED** [review]
Hi @Bantealem , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Tests passed :heavy_check_mark:
Good use gitflow :heavy_check_mark:
All linters checked green :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@codecaiine) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Bantealem** [comment]
@codecaiine  Thank you for your constructive feedback

**Pull request merged**